tests/test_contracts_type_alias_abstract_network.py

```diff
@@ -0,0 +1,24 @@
+from __future__ import annotations
+
+# This creates a type alias, to save us typing "Union[int, tuple]" everywhere
+NodeAddress = int
+
+
+###############################################################################
+# The Node class
+###############################################################################
+class Node:
+    address: NodeAddress
+
+    def __init__(self, address: NodeAddress) -> None:
+        self.address = address
+
+
+###############################################################################
+# The AbstractNetwork class
+###############################################################################
+class AbstractNetwork:
+    _nodes: dict[NodeAddress, Node]
```

Comment:
@Bruce-8 change this to `Dict` from `typing` (this was required in older versions of Python for type annotations for specifying the element types of collections)